server/src/main/java/org/elasticsearch/cluster/metadata/MetadataCreateIndexService.java

```diff
@@ -309,16 +307,13 @@ public ClusterState execute(ClusterState currentState) throws Exception {
                     aliasValidator.validateAlias(alias, request.index(), currentState.metadata());
                 }
 
-                // we only find a template when its an API call (a new index)
-                // find templates, highest order are better matching
-                List<IndexTemplateMetadata> templates =
-                        MetadataIndexTemplateService.findTemplates(currentState.metadata(), request.index());
-
-
-                Map<String, AliasMetadata> templatesAliases = new HashMap<>();
-
-                List<String> templateNames = new ArrayList<>();
-
+                // No template handling here since neither of usages of this service is relevant to them:
+                // 1. Create non-partitioned tables, templates are not related. Partitioned tables are handled by MetadataIndexTemplateService.
+                // 2. Resize partitioned tables. findTemplates always returns an empty list since
+                //    request.index() has prefix "resized" which doesn't match table pattern. See testShrinkShardsOfPartition
```

Comment:
>Is this true?

Not sure, came to this conclusion after reading the code:

 _We can change number of shards only on a specific partition_:
 
https://github.com/crate/crate/blob/master/server/src/main/java/io/crate/execution/ddl/tables/AlterTableOperation.java#L218-L221. Maybe should add "for partitioned tables" at the beginning? I meant partitioned tables here, since for regular tables templates are irrelevant.

_create another partition/index with specified size_:

https://github.com/crate/crate/blob/master/server/src/main/java/io/crate/execution/ddl/tables/AlterTableOperation.java#L243 

>Aren't templates non-relevant on resize

Yes, they are irrelevant and they are also irrelevant for other scenarios. Intention was to note scenarios where it might look like we need templates but actually we don't need it. At least for me was not obvious, had too lookup the code for all listed points. 

But if this causes confusion maybe let's drop the comment completely here and from the commit message?